tfbuilder/cases.go

```diff
@@ -0,0 +1,140 @@
+	providerFactory map[string]func() (tfprotov6.ProviderServer, error),
+	builder *Builder,
+	mesh *MeshBuilder,
+) resource.TestCase {
+	return resource.TestCase{
+		ProtoV6ProviderFactories: providerFactory,
+		Steps: []resource.TestStep{
+			{
+				Config: builder.AddMesh(mesh).Build(),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(builder.ResourceAddress("mesh", mesh.ResourceName), plancheck.ResourceActionCreate),
+					},
+				},
+			},
+			CheckReapplyPlanEmpty(builder),
+			{
+				Config: builder.AddMesh(mesh.AddToSpec("at the end", `
+  constraints = {
+    dataplane_proxy = {
+      requirements = [ { tags = { key = "a" } } ]
+      restrictions = []
+    }
+  }
+  routing = {
+    default_forbid_mesh_external_service_access = true
+  }
+`)).Build(),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(builder.ResourceAddress("mesh", mesh.ResourceName), plancheck.ResourceActionUpdate),
+						plancheck.ExpectKnownValue(builder.ResourceAddress("mesh", mesh.ResourceName), tfjsonpath.New("routing").AtMapKey("default_forbid_mesh_external_service_access"), knownvalue.Bool(true)),
+					},
+				},
+			},
+			{
+				Config: builder.AddMesh(mesh.RemoveFromSpec(`default_forbid_mesh_external_service_access = true`)).Build(),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(builder.ResourceAddress("mesh", mesh.ResourceName), plancheck.ResourceActionUpdate),
+						plancheck.ExpectKnownValue(builder.ResourceAddress("mesh", mesh.ResourceName), tfjsonpath.New("routing").AtMapKey("default_forbid_mesh_external_service_access"), knownvalue.Null()),
+					},
+				},
+			},
+			CheckReapplyPlanEmpty(builder),
+			{
+				Config: builder.AddMesh(mesh.UpdateSpec(`requirements = [ { tags = { key = "a" } } ]`, `requirements = []`)).Build(),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(builder.ResourceAddress("mesh", mesh.ResourceName), plancheck.ResourceActionUpdate),
+						plancheck.ExpectKnownValue(builder.ResourceAddress("mesh", mesh.ResourceName), tfjsonpath.New("constraints").AtMapKey("dataplane_proxy").AtMapKey("requirements"), knownvalue.ListExact([]knownvalue.Check{})),
+					},
+				},
+			},
+			CheckReapplyPlanEmpty(builder),
+			{
+				Config: builder.RemoveMesh(mesh.MeshName).Build(),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(builder.ResourceAddress("mesh", mesh.ResourceName), plancheck.ResourceActionDestroy),
+					},
+				},
+			},
+			CheckReapplyPlanEmpty(builder),
+		},
+	}
+}
+
+func CreatePolicyAndModifyFieldsOnIt(
+	providerFactory map[string]func() (tfprotov6.ProviderServer, error),
+	builder *Builder,
+	mesh *MeshBuilder,
+	mtp *PolicyBuilder,
```

Comment:
why do we need 3 different builders? is it possible to write `CreatePolicyAndModifyFieldsOnIt` so it has access only to  `mtp.AddToSpec`?